build/cmake/LLDLinker.cmake

```diff
@@ -0,0 +1,60 @@
+#[[
+    This module conditionally enables -fuse-ld=lld if it is supported by the compiler.
+    This is purely for build performance, and has no apparent effect on the generated
+    code.
+
+    LLD is *significantly* faster to link and produces significantly better link-time
+    error diagnostics.
+
+    LLD linking will be automatically enabled if it is supported, but can be forced
+    on or off by setting the MONGO_USE_LLD CMake option to ON or OFF.
+]]
+
+if(NOT COMMAND add_link_options)
+    # This only works on new-enough versions of CMake that support LINK_OPTIONS as
+    # a separate configuration entity
+    return()
+endif()
+
+include(CMakePushCheckState)
+include(CheckCSourceCompiles)
+cmake_push_check_state(RESET)
```

Comment:
Oh this is neat. I was not aware of `cmake_push_check_state` and `cmake_pop_check_state`.